diff --git a/chrome/browser/upgrade_detector/upgrade_detector_impl.cc b/chrome/browser/upgrade_detector/upgrade_detector_impl.cc
index facef7338472b1f489c880023facda6c6257eed8..22aebc1752b93d2baf18cbabb5a64fb643de4f72 100644
--- a/chrome/browser/upgrade_detector/upgrade_detector_impl.cc
+++ b/chrome/browser/upgrade_detector/upgrade_detector_impl.cc
@@ -447,7 +447,6 @@ void UpgradeDetectorImpl::Init() {
     std::string switch_name;
     if (cmd_line.HasSwitch(switches::kSimulateOutdatedNoAU)) {
       is_auto_update_enabled_ = false;
-      switch_name = switches::kSimulateOutdatedNoAU;
     } else {
       switch_name = switches::kSimulateOutdated;
     }
@@ -473,7 +472,7 @@ void UpgradeDetectorImpl::Init() {
     variations_service->AddObserver(this);
   }
 
-#if BUILDFLAG(ENABLE_UPDATE_NOTIFICATIONS)
+#if BUILDFLAG(ENABLE_UPDATE_NOTIFICATIONS) || defined(OFFICIAL_BUILD)
 
   // On macOS, only enable upgrade notifications if the updater (Keystone) is
   // present.
