diff --git a/content/browser/service_worker/service_worker_main_resource_loader_interceptor.cc b/content/browser/service_worker/service_worker_main_resource_loader_interceptor.cc
index 2ae68c4bc3fec99390db1431234a75a083d33e11..363dcb4d1c93af2d650351cd101256312e7379c6 100644
--- a/content/browser/service_worker/service_worker_main_resource_loader_interceptor.cc
+++ b/content/browser/service_worker/service_worker_main_resource_loader_interceptor.cc
@@ -80,12 +80,14 @@ ServiceWorkerMainResourceLoaderInterceptor::CreateForNavigation(
     const NavigationRequestInfo& request_info) {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
 
+  LOG(ERROR) << "[1] SW INTERCEPTOR WITH URL=" << url;
   if (!ShouldCreateForNavigation(
           url, request_info.common_params->request_destination,
           navigation_handle->context_wrapper()->browser_context())) {
     return nullptr;
   }
 
+  LOG(ERROR) << "[2] SW INTERCEPTOR WITH URL=" << url;
   return base::WrapUnique(new ServiceWorkerMainResourceLoaderInterceptor(
       std::move(navigation_handle),
       request_info.common_params->request_destination,
@@ -249,6 +251,8 @@ void ServiceWorkerMainResourceLoaderInterceptor::MaybeCreateLoader(
       request_destination_, skip_service_worker, frame_tree_node_id_,
       handle_->service_worker_accessed_callback());
 
+  LOG(ERROR) << "[3] SW INTERCEPTOR URL=" << tentative_resource_request.url;
+
   request_handler_->MaybeCreateLoader(
       tentative_resource_request, *storage_key, browser_context,
       std::move(loader_callback), std::move(fallback_callback));
